[docs] update `Add gestures` to use RNGH 2 API (#25356)
expo/expo: docs/pages/tutorial/gestures.mdx

Creating a shared value using the `useSharedValue()` hook has many advantages. It helps to mutate a piece of data and allows running animations based on the current value.
A shared value can be accessed and modified using the `.value` property. It will scale the initial value of `scaleImage` so that when a user double-taps the sticker,

> it scales to twice its original size. To do this, we will create an and call it `doubleTap`, and this object will use the `Gesture.Tap()` to animate the transition while scaling the sticker image. The `numberOfTaps` method determines how many taps are required.

Reviewer: ```suggestion
it scales to twice its original size. To do this, we will create an object and call it `doubleTap`, and this object will use the `Gesture.Tap()` to animate the transition while scaling the sticker image. The `numberOfTaps` method determines how many taps are required.
```

Not sure what wast the intended word here, but it's definitely missing

  // ...rest of the code remains same
  return (
    <View style={{ top: -350 }}>

>       /* @info Wrap the AnimatedImage component with GestureDetector*/ <GestureDetector gesture={doubleTap}>/* @end */

Reviewer 2: ```suggestion
      /* @info Wrap the AnimatedImage component with GestureDetector */<GestureDetector gesture={doubleTap}>/* @end */
```

Author: Fixed in 403c5aa

> A pan gesture allows recognizing a dragging gesture and tracking its movement. We will use this gesture handler to drag the sticker across the image.

Reviewer: ```suggestion
A pan gesture allows recognizing a dragging gesture and tracking its movement. We will use it to drag the sticker across the image.
```

A pan gesture allows recognizing a dragging gesture and tracking its movement. We will use this gesture handler to drag the sticker across the image.

> In the **EmojiSticker.js**, create an `<AnimatedView>` component using the `Animated.createAnimatedComponent()` method. Then use it to wrap the `<GestureDetector>` component by replacing the `<View>` component.

Reviewer 2: Let's use built-in `Animated.View` from Reanimated instead of calling `Animated.createAnimatedComponent(View)` manually.

Also, do we want to use `Animated.Image` as well or shall we stick with `AnimatedImage` (so that we can substitute it with `Image` from `expo-image` for which Reanimated doesn't provide a built-in wrapped component)?

Author: It has some didactic value to show `Animated.createAnimatedComponent()`but I agree that it shouldn't be used on built-in `View` and `Image` components. They are exported from Reanimated under `Animated.View` and `Animated.Image` which is both shorter and simpler to use. 

cc @amandeepmittal what do you think?

Reviewer 3: I think this is a good idea! I wasn't aware that Reanimated can export `View` and `Image` components. Can we stick to using `Image` from React Native instead of Expo Image since Expo Image doesn't work on Snack and running the examples from this tutorial is an integral part of this experience.

In the `useSharedValue()` hooks, we have set both translation variables to have an initial position of `0`.
This means that the position the sticker is initially placed is considered the starting point. This value sets the initial position of the sticker when the gesture starts.

In the previous step, we triggered the `onStart()` callback for the tap gesture chained to the `Gesture.Tap()` method.
Similarly, for the pan gesture, we have to specify an `onChange()` callback which runs when the gesture is active and is moving.

Create a `drag` object to handle the pan gesture.

{/* prettier-ignore */}
```jsx EmojiSticker.js
  const drag = Gesture.Pan()
    .onChange((event) => {
      translateX.value += event.changeX;
      translateY.value += event.changeY;
    });
```

> The `onChange()` callback accepts `event` as parameter. `changeX` and `changeY` properties hold the change in position since the last event. Use them to update the values stored in `translateX` and `translateY`.

Reviewer 3: ```suggestion
The `onChange()` callback accepts `event` as a parameter. `changeX` and `changeY` properties hold the change in position since the last event. They are used to update the values stored in `translateX` and `translateY`.
```
